### PR TITLE
Allow Umbra::Publisher to be configured

### PR DESCRIPTION
**Pull request by CGA1123**
Allow setting of :max_threads, :min_threads, and :max_queue

### CODE REVIEW

File: spec/umbra/config_spec.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
   describe '.default' do
     let(:config) { described_class.default }
 
-    it { expect(config.publisher).to eq(Umbra::Publisher) }
+    it { expect(config.publisher).to be_a(Umbra::Publisher) }
```
Comment (revdan) [inline spec/umbra/config_spec.rb]: `be_an`? not 100% sure how i feel about that

File: lib/umbra/publisher.rb [hunk 2]
```diff
@@ -1,33 +1,30 @@
 module Umbra
-  class Publisher < SynchronousPublisher
-    MAX_QUEUE_SIZE = 100
+  class Publisher
+    DEFAULT_MAX_QUEUE = 100
+    DEFAULT_MIN_THREADS = 1
+    DEFAULT_MAX_THREADS = 1
```
Comment (revdan) [inline lib/umbra/publisher.rb]: i wonder if this should be in config/env vars to be more useful

File: spec/umbra/config_spec.rb (hunk 1 above)
Comment (CGA1123) [reply spec/umbra/config_spec.rb]: 🤔 Any suggestions on a better way about this. The only expectation is that the default config is sensible and works.

File: lib/umbra/publisher.rb (hunk 2 above)
Comment (CGA1123) [reply lib/umbra/publisher.rb]: Not sure what you mean by this, this change enables users to configure all of these via
```ruby
Umbra.configure do |config|
  config.publisher = Umbra::Publisher.new(max_treads: ENV['UMBRA_MAX_THREADS'])
end
```

Do you mean having the ability to configure this purely via passing an options hash to `Umbra.configure`? I guess something like that would be similar to how `Rails.cache` is configured 🤔

Comment (revdan) [reply lib/umbra/publisher.rb]: nope, misread it. all good 👍

**Pull request merged**